numba/unicode.py

```diff
@@ -622,6 +622,29 @@ def join_str_impl(sep, parts):
         return join_str_impl
 
 
+@overload_method(types.UnicodeType, 'zfill')
+def unicode_zfill(str_to_fill, width_to_fill):
+    def zfill_impl(fstr, fwidth):
+
+        fstr_len = len(fstr)
+
+        if fwidth <= fstr_len or fstr_len == 0:
```

Comment:
What happens if the argument is negative? This is a bit speculative, but can you simply write: `if fwidth <= fstr_len` ? That will also short-path the case when the argument is negative.